From e33c0e68c950e51f6cc0f85f71c2648e80900f89 Mon Sep 17 00:00:00 2001
From: Wenjing Liu <Wenjing.Liu@amd.com>
Date: Mon, 21 Aug 2017 16:43:08 -0400
Subject: [PATCH 3190/5855] drm/amd/display: retry LT on failing LT in
 automated request

[Description]
Link training is not retried in automated link training request.
In case link training is occasionally failing, there is not way to retry
so the requested lane settings will not set.

Change-Id: I46fbdcf9d115bb5d8a841134e5db2c95ef81ee3f
Signed-off-by: Wenjing Liu <Wenjing.Liu@amd.com>
Reviewed-by: Tony Cheng <Tony.Cheng@amd.com>
Acked-by: Harry Wentland <Harry.Wentland@amd.com>
---
 drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c b/drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c
index 3a5abd9..23d8b61 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc_link_hwss.c
@@ -299,10 +299,11 @@ void dp_retrain_link_dp_test(struct dc_link *link,
 
 			dp_receiver_power_ctrl(link, true);
 
-			dc_link_dp_perform_link_training(
+			perform_link_training_with_retries(
 					link,
 					link_setting,
-					skip_video_pattern);
+					skip_video_pattern,
+					LINK_TRAINING_ATTEMPTS);
 
 			link->cur_link_settings = *link_setting;
 
-- 
2.7.4

